app/src/test/java/app/wefridge/wefridge/ItemControllerUnitTest.kt

```diff
@@ -1,154 +1,129 @@
-                "contact_name" to contactName,
-                "contact_email" to contactEmail,
-                "owner" to ownerRef
-            )
-        }
-    }
 
-    @Test
-    fun testParsingWithAllFieldsExisting() {
-        val parsedItem = ItemController.parse(itemData, firebaseId)
-
-        val expectedItem = Item(firebaseId, name, description, isShared, quantity.toInt(), unit, sharedEmail, bestByDate, location, geohash, contactName, contactEmail, ownerRef)
-        assertEquals(expectedItem, parsedItem)
-    }
+class ItemControllerUnitTest {
 
-    @Test
-    fun testParsingWithoutFirebaseId() {
-        val parsedItem = ItemController.parse(itemData, null)
-        val expectedItem = Item(null, name, description, isShared, quantity.toInt(), unit, sharedEmail, bestByDate, location, geohash, contactName, contactEmail, ownerRef)
-        assertEquals(expectedItem, parsedItem)
+    // TODO: test implementation of listeners in deleteItem (when related branch gets pulled into main)
+    // TODO: consider to delete getItems in ItemController
+
+    // Mock objects have the same methods as the original class
+    private val itemDocSnap = Mockito.mock(DocumentSnapshot::class.java)
+    private val ownerDocRef = Mockito.mock(DocumentReference::class.java)
+
+    // dummy data
+    private var dummyUserName = "Karen Anderson"
+    private var dummyUserEmail = "karen@anderson.de"
+    private var dummyFirebaseId = "demo_id"
+    private var dummyGeoPoint = GeoPoint(59.3294,18.0686)
+    private var dummyValues = mapOf<String, Any>(
+        ITEM_OWNER to ownerDocRef,
+        ITEM_NAME to "Gouda Cheese",
+        ITEM_DESCRIPTION to "So good!",
+        ITEM_IS_SHARED to true,
+        ITEM_QUANTITY to 10L,
+        ITEM_UNIT to Unit.KILOGRAM.value.toLong(),
+        ITEM_BEST_BY to Date(),
+        ITEM_LOCATION to dummyGeoPoint,
+        ITEM_GEOHASH to GeoFireUtils.getGeoHashForLocation(GeoLocation(dummyGeoPoint.latitude, dummyGeoPoint.longitude)),
+        ITEM_CONTACT_NAME to dummyUserName,
+        ITEM_CONTACT_EMAIL to dummyUserEmail
+    )
+
+    @Before
+    fun setUp() {
+        // the following instructions regarding the mockk library were inspired by
+        // https://stackoverflow.com/questions/58158711/android-local-unit-test-mock-firebaseauth-with-mockk/58158712#58158712
+        mockkStatic(FirebaseFirestore::class)
+        every { FirebaseFirestore.getInstance() } returns mockk(relaxed = true)
     }
 
-    @Test
-    fun testParsingWithoutName() {
-        val modifiedItemData = mapWithNullValue(itemData, "name")
-        val parsedItem = ItemController.parse(modifiedItemData, firebaseId)
-        val itemExpected = Item(firebaseId, null, description, isShared, quantity.toInt(), unit, sharedEmail, bestByDate, location, geohash, contactName, contactEmail, ownerRef)
-        assertEquals(itemExpected, parsedItem)
+    @After
+    fun tearDown() {
+        setUpDocumentSnapshot(itemDocSnap, dummyValues, dummyFirebaseId)
```

Comment:
Why is this called on tearDown?

Reply:
Calling it in the setUp method would have caused 'testParsingWithoutItemOwner' to fail. I have refactored "setUpDocumentSnapshot" now, so that we get can call a modfied version of it in the setUp method.